src/libs/OptionsListUtils.js

```diff
@@ -488,12 +488,16 @@ function getOptions(reports, personalDetails, activeReportID, {
             : '';
 
         const reportContainsIOUDebt = iouReportOwner && iouReportOwner !== currentUserLogin;
+        const hasAddWorkspaceRoomError = report.errorFields && !_.isEmpty(report.errorFields.addWorkspaceRoom);
         const shouldFilterReportIfEmpty = !showReportsWithNoComments && report.lastMessageTimestamp === 0
 
                 // We make exceptions for defaultRooms and policyExpenseChats so we can immediately
                 // highlight them in the LHN when they are created and have no messsages yet. We do
                 // not give archived rooms this exception since they do not need to be higlihted.
-                && !(!ReportUtils.isArchivedRoom(report) && (isDefaultRoom || isPolicyExpenseChat));
+                && !(!ReportUtils.isArchivedRoom(report) && (isDefaultRoom || isPolicyExpenseChat))
+
+                // Also make an exception for workspace rooms that failed to be added
+                && !hasAddWorkspaceRoomError;
```

Comment:
NAB: this is definitely a complex and difficult to understand condition, I wonder if there are any ways to simplify it

Reply:
Yeah that's pretty ugly. It would help to break it out into several descriptive boolean variables. I think @tgolen might be cleaning that up with his LHN cleanup issues? I'm going to leave it as is for now.

Reply:
Also, this logic has actually been moved here in main
https://github.com/Expensify/App/blob/7fbe704ac8291944a7a8533bbcfa1df0e207b182/src/libs/SidebarUtils.js#L125-L134

Reply:
Yeah, let me take a try at cleaning up this logic 👍 I created a GH for it here: https://github.com/Expensify/Expensify/issues/229633